### PR TITLE
Update Preference Page Layout

### PR DESCRIPTION
**Pull request by ingomohr**
Hey @AObuchow,

as discussed yesterday, this is the PR for the layout-update of the preference page.

There are still some things I would like to fix/discuss - maybe not all have to be processed w/ this PR.

* Bug: Eclipse UI doesn't refresh on moving the scale-selection while keeping mouse-down (only on mouse_up events)
  * I think this was different before. I think I missed some line that was there before the refactoring.
* No test fragment: For now I gave up adding a test fragment for the control
  * I didn't manage to run the (plain) unit tests for the control on Maven
  * We can still add this later.
* CSS-Scheme text widget has bright background
  * I'm not sure whether this was the case before, as well.

If you got suggestions on improvements, bugs or other feedback, I'd be glad of course to hear from you. ;)

### CODE REVIEW

**Comment by ingomohr** [comment]
##### Refresh-on-Close
There's another thing (which was this way before, too - if I don't miss a thing)

At the moment the scales are updated when the pref page is closed (cancel or ok).
This fails because the page is already disposed w/ its widgets.
I think we could improve this by only refreshing the widgets while the pref page is visible.

File: com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/PreferencesPage.java [hunk 1]
```diff
@@ -109,26 +76,20 @@ private void addHSLCustomization(Group customizeGroup) {
 			colorManager.setBaseColor(newBaseColor.getColor());
 			colorManager.setAccentColor(newAccentColor.getColor());
 			colorManager.saveColors();
+			System.out.println("save hue");
```
Comment (AObuchow) [inline com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/PreferencesPage.java:79]: Remove before merge ;)

File: com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+		addDisposeListener(e -> {
+			if (kit != null) {
+				kit.dispose();
+			}
+		});
+	}
+
+	private void create() {
+
```
Comment (AObuchow) [inline com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java:56]: Remove whitespace

File: com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		setLayout(GridLayoutFactory.fillDefaults().create());
+
```
Comment (AObuchow) [inline com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java:58]: Remove whitespace

**Comment by AObuchow** [comment]
Thanks for the great PR Ingo :D This is quite a nice refactoring, so much cleaner.
Aside from the minor comments, this looks good to merge IMO.

> There are still some things I would like to fix/discuss - maybe not all have to be processed w/ this PR.
> 
>     * Bug: Eclipse UI doesn't refresh on moving the scale-selection while keeping mouse-down (only on mouse_up events)
>       
>       * I think this was different before. I think I missed some line that was there before the refactoring.

The behavior seems to be the same as it was pre-refactoring, in my quick testing. 
Do you mean the mouse has to stop moving for the refresh to occur? If so, this was already an issue I believe.

> 
>     * No test fragment: For now I gave up adding a test fragment for the control
>       
>       * I didn't manage to run the (plain) unit tests for the control on Maven
>       * We can still add this later.

This is totally okay :) If you're willing to, please open a bug about it just so we can track adding tests as an issue.

> 
>     * CSS-Scheme text widget has bright background
>       
>       * I'm not sure whether this was the case before, as well.

This indeed wasn't the case before, but on Linux it loosk OK to me:
image

Does it look considerably worst on OSX? Could you share a photo?

I think the styling of this element might be an actual bug in Spectrum Theme's CSS :o it might have to do with the ExpandableComposite or Twistie's styling (or lack of).
We can open up a separate bug about this, the CSS fix should be quite simple.

**Comment by AObuchow** [comment]
> ##### Refresh-on-Close
> 
> There's another thing (which was this way before, too - if I don't miss a thing)
> 
> At the moment the scales are updated when the pref page is closed (cancel or ok).
> This fails because the page is already disposed w/ its widgets.
> I think we could improve this by only refreshing the widgets while the pref page is visible.

Agreed it'd be nice to fix this. Maybe the fix would be from adding a check in each scales selection listener for the widget being disposed, or dispose all selection listeners that are associated with the scales when the preference page is disposed?

File: com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/PreferencesPage.java (hunk 1 above)
Comment (ingomohr) [reply com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/PreferencesPage.java:79]: Ouh nou! 😇 . Yes, those gotto go. Thank you for finding. :)

Comment (AObuchow) [reply com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/PreferencesPage.java:79]: No worries :D

Comment (ingomohr) [reply com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/PreferencesPage.java]: Fixed in latest push (amended commit)

File: com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java (hunk 2 above)
Comment (ingomohr) [reply com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java]: Fixed in latest push (amended commit)

File: com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java (hunk 3 above)
Comment (ingomohr) [reply com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/SpectrumPreferencesControl.java]: Fixed in latest push (amended commit)

**Comment by ingomohr** [comment]
> Thanks for the great PR Ingo :D This is quite a nice refactoring, so much cleaner.
> Aside from the minor comments, this looks good to merge IMO.
> 
> > There are still some things I would like to fix/discuss - maybe not all have to be processed w/ this PR.
> > ```
> > * Bug: Eclipse UI doesn't refresh on moving the scale-selection while keeping mouse-down (only on mouse_up events)
> >   
> >   * I think this was different before. I think I missed some line that was there before the refactoring.
> > ```
> 
> The behavior seems to be the same as it was pre-refactoring, in my quick testing.
> Do you mean the mouse has to stop moving for the refresh to occur? If so, this was already an issue I believe.
> 
> > ```
> > * No test fragment: For now I gave up adding a test fragment for the control
> >   
> >   * I didn't manage to run the (plain) unit tests for the control on Maven
> >   * We can still add this later.
> > ```
> 
> This is totally okay :) If you're willing to, please open a bug about it just so we can track adding tests as an issue.
> 
> > ```
> > * CSS-Scheme text widget has bright background
> >   
> >   * I'm not sure whether this was the case before, as well.
> > ```
> 
> This indeed wasn't the case before, but on Linux it loosk OK to me:
> image
> 
> Does it look considerably worst on OSX? Could you share a photo?
> 
> I think the styling of this element might be an actual bug in Spectrum Theme's CSS :o it might have to do with the ExpandableComposite or Twistie's styling (or lack of).
> We can open up a separate bug about this, the CSS fix should be quite simple.

This is what it looks like on MacOS. Looks ok to me.

Screenshot 2020-08-27 at 23 20 03

**Comment by ingomohr** [comment]
> > ##### Refresh-on-Close
> > There's another thing (which was this way before, too - if I don't miss a thing)
> > At the moment the scales are updated when the pref page is closed (cancel or ok).
> > This fails because the page is already disposed w/ its widgets.
> > I think we could improve this by only refreshing the widgets while the pref page is visible.
> 
> Agreed it'd be nice to fix this. Maybe the fix would be from adding a check in each scales selection listener for the widget being disposed, or dispose all selection listeners that are associated with the scales when the preference page is disposed?

Yes, maybe disposing the listeners is a good option. Actually, the entire preference page is (I think) being disposed. So, we should drop the listeners, as well. 👍

**Comment by ingomohr** [comment]
Thank you for the review, Andrew! 😄 . And sorry for the ``Syso`` 😇.

**Comment by ingomohr** [comment]
I too think we maybe can improve the points we spotted with extra little issues. I'd be glad to help you there if you'll have me (next week). :)

**Comment by AObuchow** [comment]
> Yes, maybe disposing the listeners is a good option. Actually, the entire preference page is (I think) being disposed. So, we should drop the listeners, as well. +1

Ok sounds good, I can make a bug about this :)

**Comment by AObuchow** [comment]
> Thank you for the review, Andrew! smile . And sorry for the `Syso` innocent.

My pleasure, thanks for the great PR and no worries, I do that all the time ;) Merging!

**Pull request merged**